### DISCUSSION TITLE
React Beautiful DnD with v6 Field Array

### DISCUSSION THREAD

**Discussion by nlrowe** [react-hook-form/react-hook-form#2309] [Q&A]
After upgrading to v6 field array value changes no longer persist after a drag and drop using React Beautiful DnD.

https://codesandbox.io/s/react-hook-form-usefieldarray-kit8r?file=/src/index.js

**Steps to Reproduce:**
1. Enter a value in the two inputs shown
2. Using the 'Drag Me' element drop one input into the place of the other

**Result:**
Data entered into both fields are reset to the default
**Expected Result:**
Data entered moves with the input

Switch React Hook Form version to 5.7.2 in the code sandbox and repeat the steps. The changed values move as expected.

Any idea what has changed?

**Comment by bluebill1049** [comment]
hmmm interesting! I will take a look at this.

**Reply by dpw1** [reply]
Hey!

Any updates on this? I'm also dealing with the same issue.

Thank you!

**Reply by bluebill1049** [reply]
i think this is patched right: https://codesandbox.io/s/react-hook-form-usefieldarray-forked-1tzyn?file=/src/index.js sorry forgot to follow it up.
